pkg/cluster/kubernetes/sync.go

```diff
@@ -239,6 +240,9 @@ func (c *Cluster) getAllowedResourcesBySelector(selector string) (map[string]*ku
 	}
 
 	for _, resource := range resources {
+		if resource == nil {
```

Comment:
Why is this needed?

Reply:
To avoid panics, a resource can be nil. I've hit this while testing it, I can remove it if you think it's not worth it.

Reply:
The client can return nil entries? Eugh. OK, that's sensible then -- maybe signpost it with a brief comment.